src/components/DeviceTile.tsx

```diff
@@ -13,9 +13,10 @@ type Props = {
   id: string;
 };
 
-const getDeviceType = (stream: MediaStream) => {
-  if (stream.getVideoTracks().length > 0) {
-    return "video";
+const getDeviceType = (streamInfo: StreamInfo) => {
+  if (streamInfo.stream.getVideoTracks().length > 0) {
+    if (streamInfo.id.includes("screen")) return "screenshare";
```

Comment:
it's some kind of magic. You could introduce another field `trackType: "camera" | "mock" | "screen share" | "audio"` because relaying on id is risky